### PR TITLE
osm-0.4.0

### PR DESCRIPTION
**Pull request by EItanya**
BOT NOTES: 
resolves https://github.com/solo-io/service-mesh-hub/issues/987

### CODE REVIEW

**Comment by joekelley** [comment]
Thanks for making these changes @ashleywang1! Can you also validate our [osm docs](https://docs.solo.io/service-mesh-hub/latest/guides/osm/) as part of this change, and fix any issues you find there? We may need to update `meshctl demo osm` as well.

File: ci/setup-funcs.sh [hunk 1]
```diff
@@ -275,14 +275,26 @@ function install_osm() {
...
+
 
   ${ROLLOUT} -n osm-system osm-controller
 
   for i in bookstore bookthief bookwarehouse bookbuyer; do ${K} create ns $i; done
 
-  for i in bookstore bookthief bookwarehouse bookbuyer; do osm namespace add $i; done
+  if [[ "$VERSION" == *"$V3"* ]]; then
+    for i in bookstore bookthief bookwarehouse bookbuyer; do osm namespace add $i; done
+  else
```
Comment (harveyxia) [inline ci/setup-funcs.sh:294]:
```suggestion
  else
  # for OSM versions >= v0.4.0
```

File: pkg/meshctl/commands/demo/common/initialize/scripts/install_osm.sh [hunk 2]
```diff
@@ -14,7 +14,13 @@ ${ROLLOUT} -n osm-system osm-controller
 
 for i in bookstore bookthief bookwarehouse bookbuyer; do ${K} create ns $i; done
 
-for i in bookstore bookthief bookwarehouse bookbuyer; do osm namespace add $i; done
+VERSION=$(osm version)
+V3='v0.3.0'
+if [[ "$VERSION" == *"$V3"* ]]; then
+  for i in bookstore bookthief bookwarehouse bookbuyer; do osm namespace add $i; done
+else
```
Comment (harveyxia) [inline pkg/meshctl/commands/demo/common/initialize/scripts/install_osm.sh:21]:
```suggestion
else
# for OSM versions >= v0.4.0
```

File: docs/content/guides/osm/_index.md [hunk 3]
```diff
@@ -77,6 +77,16 @@ kubectl rollout status deployment --timeout 300s -n bookwarehouse bookwarehouse
 kubectl rollout status deployment --timeout 300s -n bookbuyer bookbuyer
 ```
 
+{{% notice note %}}
+For osm version v0.3.0, the flag --enable-sidecar-injection doesn't exist, so you will want to use:
```
Comment (harveyxia) [inline docs/content/guides/osm/_index.md:81]: Can we also add a note towards the top of the guide stating that the guide assumes OSM v0.4.0?

**Review by harveyxia — APPROVED** [review]

**Pull request merged**